lib/puppet/provider/user/hpux.rb

```diff
@@ -63,4 +75,43 @@ def password
       return ent.passwd
     end
   end
+
+  def trusted
+      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
+      if trusted_sys.chomp == "System is not trusted."
+         "NotTrusted"
+      else
+         "Trusted"
+      end
+  end
+
+  def trust2
```

Comment:
I will look it over again. I am just not able to get any useful data out of
the ruby and spec scripts as to what is going on under the covers.  The
only way to know if the box is trusted is to go out and check for it. I
have tried multiple ways to get it to work without having the command there
and nothing seems to work.

On Mon, Jan 5, 2015 at 2:33 PM, Michael Smith <notifications@github.com>
wrote:

> In lib/puppet/provider/user/hpux.rb
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22483439>:
>
> > @@ -63,4 +75,43 @@ def password
> >        return ent.passwd
> >      end
> >    end
> > +
> > +  def trusted
> > +      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
> > +      if trusted_sys.chomp == "System is not trusted."
> > +         "NotTrusted"
> > +      else
> > +         "Trusted"
> > +      end
> > +  end
> > +
> > +  def trust2
>
> If trust2 is purely for testing, I would move it's implementation to the
> spec test. Simpler interfaces are easier to maintain.
>
> I would look at
> https://github.com/puppetlabs/puppet/blob/master/docs/rspec_tutorial.md
> for an introduction to how we write specs. There are many examples in
> spec/unit/* once you've read the tutorial. The provider/user/hpux_spec.rb
> that you modified has examples of using the #stub method, which are what
> you want to emulate.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22483439>.
>